packages/jest-reporters/src/GitHubActionsReporter.ts

```diff
@@ -6,54 +6,79 @@
  */
 
 import stripAnsi = require('strip-ansi');
-import type {
-  AggregatedResult,
-  TestContext,
-  TestResult,
-} from '@jest/test-result';
+import type {Test, TestResult} from '@jest/test-result';
+import type {Config} from '@jest/types';
+import {
+  formatPath,
+  getStackTraceLines,
+  getTopFrame,
+  separateMessageFromStack,
+} from 'jest-message-util';
 import BaseReporter from './BaseReporter';
 
-const lineAndColumnInStackTrace = /^.*?:([0-9]+):([0-9]+).*$/;
+type AnnotationOptions = {
+  file?: string;
+  line?: number | string;
+  message: string;
+  title: string;
+  type: 'error' | 'warning';
+};
 
-function replaceEntities(s: string): string {
-  // https://github.com/actions/toolkit/blob/b4639928698a6bfe1c4bdae4b2bfdad1cb75016d/packages/core/src/command.ts#L80-L85
-  const substitutions: Array<[RegExp, string]> = [
-    [/%/g, '%25'],
-    [/\r/g, '%0D'],
-    [/\n/g, '%0A'],
-  ];
-  return substitutions.reduce((acc, sub) => acc.replace(...sub), s);
-}
+const titleSeparator = ' \u203A ';
 
 export default class GitHubActionsReporter extends BaseReporter {
   static readonly filename = __filename;
 
-  override onRunComplete(
-    _testContexts?: Set<TestContext>,
-    aggregatedResults?: AggregatedResult,
-  ): void {
-    const messages = getMessages(aggregatedResults?.testResults);
+  onTestFileResult({context}: Test, {testResults}: TestResult): void {
```

Comment:
`onTestFileResult` is needed to annotate retries. I was trying to use `onTestCaseResult` first, but it reports retried tests as failures. (By the way, strangely `TestContext` from `onTestCaseResult` has wrong `rootDir`, but all is good with `onTestFileResult`.)

Reply:
> By the way, strangely `TestContext` from `onTestCaseResult` has wrong `rootDir`, but all is good with `onTestFileResult`.)

can you file an issue for it?

Reply:
Have to figure out what is the root of that. Locally all was fine, but not on CI. Strange thing. Let’s see.